File: DS_A/Python/Dynamic_Programming/fibonacci_v1.py
a = int(input())

baza = [-1]*(a+1)

def f(a:int) -> int:
  if a == 0: return 1 
  if a == 1: return 1 
  
  if baza[a] != -1: return a
  
  return f(a-1) + f(a-2)

print(f(a-1))
  

File: DS_A/Dynamic_Programming/fibonacci_v1.cpp
// Bu usul rekursiyaga asoslanib yechilgan!

#include <iostream>
#include <vector>

#define ll long long
using namespace std;

vector <ll> baza;

ll f(ll a){
  if(a==0) return 0;
  if(a==1) return 1;
  
  if(baza[a]!=-1) return a;
  
  return f(a-1) + f(a-2);
}

int main() 
{
    ll a;
    cin>>a;
    baza.resize(a+1,-1);
    cout<<f(a);
}
